common/src/main/java/com/wynntils/gui/render/FontRenderer.java

```diff
@@ -48,7 +50,10 @@ public void renderText(
 
         if (text == null) return;
 
-        // TODO: Add rainbow color support
+        if (customColor == CommonColors.RAINBOW) {
```

Comment:
If you call this after the render coord calculations, you can remove duplicate code